crates/rpc/rpc-types-compat/src/transaction.rs

```diff
@@ -46,3 +50,86 @@ pub trait TransactionCompat: Send + Sync + Unpin + Clone + Debug {
         request: TransactionRequest,
     ) -> Result<TxTy<Self::Primitives>, Self::Error>;
 }
+
+/// Converts `self` into `T`.
+pub trait IntoRpcTx<T> {
+    /// Performs the conversion.
+    fn into_rpc_tx(self, tx_info: TransactionInfo) -> T;
+}
+
+/// Converts [`TransactionRequest`] into `Self`.
+pub trait FromTxReq {
```

Comment:
should we rename this `TryIntoSimulated` because this is specifically for eth_simulateV1